pkg/pillar/types/zedagenttypes.go

```diff
@@ -488,6 +488,7 @@ type ZedAgentStatus struct {
 	MaintenanceMode      bool       // Don't run apps etc
 	ForceFallbackCounter int        // Try image fallback when counter changes
 	CurrentProfile       string     // Current profile
+	BaseosUpdateCounter  uint32     // Current BaseosUpdateCounter from config
```

Comment:
Can we 1) make this a BaseOs struct (which will later also have a content tree UUID) and 2) rename the counter inside that struct to ConfigRetryUpdateCounter

Reply:
Done, but I still have checking for nil of BaseOs in parseconfig, please take a look.